OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyUtils.cs

```diff
@@ -246,7 +246,7 @@ public static string DescriptiveIpAddress(string ip)
 		{
 			if (ip == null)
 				return "Unknown Host";
-			if (ip == IPAddress.Loopback.ToString())
+			if (ip == "localhost")
```

Comment:
You have to do a reverse dns lookup to catch all localhost addresses. The parameter will always be an IP address instead of a domain name with the current codebase.

Reply:
The reverse lookup can stall for hundreds of milliseconds (see comment of pchote above) if it doesn't match localhost. In this case it is a recurring effect which means that it must be done asynchronously.